test/extensions/common/sqlutils/sqlutils_test.cc

```diff
@@ -0,0 +1,164 @@
+// Under each table there is secondary list which contains operations performed
+// on the table, like "select", "insert", etc.
+TEST_P(MetadataFromSQLTest, ParsingAndMetadataTest) {
+  // Get the SQL query
+  const std::string& query = std::get<0>(GetParam());
+  ProtobufWkt::Struct metadata;
+
+  // Check if the parsing result is what expected.
+  ASSERT_EQ(std::get<1>(GetParam()), SQLutils::setMetadata(query, metadata));
+
+  // If parsing was expected to fail do not check parsing values.
+  if (!std::get<1>(GetParam())) {
+    return;
+  }
+
+  // Access metadata fields, where parsing results are stored.
+  auto& fields = *metadata.mutable_fields();
+
+  // Get the names of resources which SQL query operates on.
+  std::map<std::string, std::list<std::string>> expected_tables = std::get<2>(GetParam());
+  // Check if query results return the same number of resources as expected.
+  ASSERT_EQ(expected_tables.size(), fields.size());
+  for (const auto& i : fields) {
+    // Get from created metadata the list of operations on the resource
+    const auto& operations = i;
+    // Get the list of expected operations on the same resource from test param.
+    const auto& table_name_it = expected_tables.find(operations.first);
+    // Make sure that a resource (table) found in metadata is expected.
+    ASSERT_NE(expected_tables.end(), table_name_it);
+    auto& operations_list = table_name_it->second;
+    // The number of expected operations and created in metadata must be the same.
+    ASSERT_EQ(operations_list.size(), operations.second.list_value().values().size());
+    // Now iterate over the operations list found in metadata and check if the same operation
+    // is listed as expected in test param.
+    for (const auto& j : operations.second.list_value().values()) {
+      // Find that operation in test params.
+      const auto operation_it =
+          std::find(operations_list.begin(), operations_list.end(), j.string_value());
+      ASSERT_NE(operations_list.end(), operation_it);
+      // Erase the operation. At the end of the test this list should be empty what means
+      // that we found all expected operations.
+      operations_list.erase(operation_it);
+    }
+    // Make sure that we went through all expected operations.
+    ASSERT_TRUE(operations_list.empty());
+    // Remove the table from the list. At the end of the test this list must be empty.
+    expected_tables.erase(table_name_it);
+  }
+
+  ASSERT_TRUE(expected_tables.empty());
+}
+
+#define TEST_VALUE(...)                                                                            \
+  std::tuple<std::string, bool, std::map<std::string, std::list<std::string>>> { __VA_ARGS__ }
+INSTANTIATE_TEST_SUITE_P(
+    SQLUtilsTestSuite, MetadataFromSQLTest,
+    ::testing::Values(
+        TEST_VALUE("blahblah;", false, {}),
+
+        TEST_VALUE("CREATE TABLE IF NOT EXISTS table1(Usr VARCHAR(40),Count INT);", true,
```

Comment:
Can we check for all lowercased queries?

```
create tabel if not exists table1(usr varchar(40),count int);
```

Reply:
Changed testing routine to convert queries to lowercase and uppercase, so all different variants are tested.